torchtune/modules/tied_linear.py

```diff
@@ -9,13 +9,33 @@
 import torch.nn.functional as F
 
 
+class Linear(nn.Module):
```

Comment:
docstring explains why i had to add this

Reply:
Sorry I'm a bit confused by this change on two fronts: 

(1) Does this not change the key names in the state dict?
(2) Now that we have a module again, how do we not wind up right back where we started with to_empty?

Reply:
(1) No. The Linear nn.Module doesnt have any weights. The weight is passed in the forward only. 
```
Class Linear(nn.Module):
      def forward(x, weight):
             return F.Linear(x, weight)
```

TiedLinear is still a regular python class, and the key name is still model.output.tied_module.weight --> translates to --> model.TiedLinear.tok_embedding.weight

(2) i tested it with FSDP, and confirmed that the memory pointers are the same in model.tok_embedding.weight and model.output.weight. So things are fine. Is that what you were referencing too?